R/tbl_ard_wide_summary.R

```diff
@@ -0,0 +1,144 @@
+#' Wide ARD summary table
+#'
+#' `r lifecycle::badge("experimental")`\cr
+#' This function is similar to `tbl_ard_summary()`, but places summary statistics
+#' wide, in separate columns.
+#' All included variables must be of the same summary type, e.g. all continuous
+#' summaries or all categorical summaries (which encompasses dichotomous variables).
+#'
+#' @param cards (`card`)\cr
+#'   An ARD object of class `"card"` typically created with `cards::ard_*()` functions.
+#' @inheritParams tbl_wide_summary
+#'
+#' @return a gtsummary table of class 'tbl_wide_summary'
+#' @export
+#'
+#' @examples
+#' library(cards)
+#'
+#' ard_stack(
+#'   trial,
+#'   ard_continuous(variables = age),
+#'   .missing = TRUE,
+#'   .attributes = TRUE
+#' ) |>
+#'   tbl_ard_wide_summary()
+#'
+#' ard_stack(
+#'   trial,
+#'   ard_dichotomous(variables = response),
+#'   ard_categorical(variables = grade),
+#'   .missing = TRUE,
+#'   .attributes = TRUE
+#' ) |>
+#'   tbl_ard_wide_summary()
+tbl_ard_wide_summary <- function(cards,
+                                 statistic =
+                                   switch(type[[1]],
+                                          "continuous" = c("{median}", "{p25}, {p75}"),
+                                          c("{n}", "{p}%")),
```

Comment:
Just playing with this, I got here:
```r
trial1 <- head(trial)
bind_ard(
  ard_continuous(trial1, variables = c(age, marker)),
  ard_categorical(trial1, variables = c(age, marker)),
  ard_attributes(trial1, variables = c(age, marker))
) |>
  brdg_wide_summary(
    variables = c("age", "marker"),
    statistic = list(age = c("{mean}", "{sd}", "{p}%"), marker = c("{mean}", "{sd}", "{p}%")),
    type = list(age = "continuous", marker = "continuous")
  )
```
Here only the first is called in. I was wondering if we should throw in an error or allow it with a warning or not allow it. Then would it make sense to NOT enforce the either categorical/dichotomous either numeric?